python/mspasspy/util/converter.py

```diff
@@ -480,6 +482,185 @@ def TimeSeriesEnsemble2Stream(tse):
 TimeSeriesEnsemble.toStream = TimeSeriesEnsemble2Stream
 
 
+def Pf2AttributeNameTbl(pf, tag="attributes"):
+    """
+    This function will parse a pf file to extract a tbl with a specific
+    key and return a data structure that defines the names and types of
+    each column in the input file.
+
+    The structure returned is a tuple with three components:
+        1 (index 0) python array of attribute names in the original tbl order
+          This is used to parse the text file so the order matters a lot.
+        2 (index 1) parallel array of type names for each attribute.
+          These are actual python type objects that can be used as the
+          second arg of isinstance.
+        3 (index 2) python dictionary keyed by name field that defines
+          what a null value is for this attribute.
+
+    :param pf:  AntelopePf object to be parsed
+    :param tag:  &Tbl tag for section of pf to be parsed.
+
+    """
+    tbl = pf.get_tbl(tag)
+    names = []
+    dtypes = []
+    nullvalues = []
+
+    i = 0
+    for line in tbl:
+        temp = line.split()
+        names.append(temp[0])
+        typenamein = temp[1].lower()  # allows upper or lower case in names
+        if typenamein == "string" or typenamein == "str":
+            typ = type("foobar")
+            nullval = temp[2]
+        elif typenamein == "integer" or typenamein == "int" or typenamein == "long":
+            typ = type(1)
+            nullval = int(temp[2])
+        elif (
+            typenamein == "float"
+            or typenamein == "double"
+            or typenamein == "real"
+            or typenamein == "epochtime"
+        ):
+            typ = type(1.0)
+            nullval = float(temp[2])
+        elif typenamein == "bool" or typenamein == "boolean":
+            typ = type(True)
+            nullval = bool(temp[2])
+        else:
+            raise MsPASSError(
+                "parse_attribute_name_tbl:  unsupported data type file=" + typenamein,
+                "Fatal",
+            )
+        dtypes.append(typ)
+        nullvalues.append(nullval)
+
+        i += 1
+        nulls = dict()
+        i = 0
+        for k in names:
+            nulls[k] = nullvalues[i]
+            i += 1
+
+    return tuple([names, dtypes, nulls])
+
+
+def Textfile2Dataframe(
+    filename,
+    separator="\s+",
```

Comment:
May want to state clearly that the default separator is a regular expression of "one or more spaces". We should also point the user to the corresponding documentations of the underlying read_csv calls (i.e. [this](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html))